react/week2/to_do_list/src/index.js

```diff
@@ -0,0 +1,177 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import "./index.css";
+
+const taskList = [
+  {
+    id: 0,
+    description: "Get out of bed",
+    color: "brown",
+    deadline: "2020-06-08",
+    done: false,
+    visibility: true
+  },
+  {
+    id: 1,
+    description: "Brush teeth",
+    color: "red",
+    deadline: "2020-06-08",
+    done: false,
+    visibility: true
+  },
+  {
+    id: 2,
+    description: "Eat breakfast",
+    color: "orange",
+    deadline: "2020-06-08",
+    done: true,
+    visibility: true
+  }
+];
+
+class Header extends React.Component {
+  render() {
+    return (
+      <div>
+        <h1> Welcome to My ToDo List</h1>
+        <br ></br>
+      </div>
+    )
+  }
+}
+
+class Counter extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      ellapsedTime: this.props.initial,
+    };
+  }
+
+  componentDidMount() {
+    setInterval(this.increment, 1000);
+  }
+
+  increment = () => {
+    this.setState({ ellapsedTime: this.state.ellapsedTime + 1 })
+  }
+
+  StartTimer = () => {
+    setInterval(this.increment, 1000);
+  }
+
+  render() {
+
+    return (
+      <div>
+        <p> you have spent {this.state.ellapsedTime} seconds in this page</p>
+        <br></br>
+      </div>
+    )
+  }
+}
+
+class TodoList extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      toDoListData: this.props.toDoListData
+    }
+  }
+
+  addRandomTask(toDoListData) {
+    toDoListData.push({
+      id: toDoListData.length,
+      description: "Random Text",
+      color: 'navygrey',
+      deadline: "2020-12-29",
+      done: false,
+      visibility: true
+    });
+    this.setState({ toDoListData: this.props.toDoListData })
+  }
+
+  render() {
+    let toDoListJSXarray = [];
+    for (let i = 0; i < this.props.taskList.length; i++) {
+      toDoListJSXarray.push(<ToDoTask task={this.props.taskList[i]} />)
+    }
```

Comment:
I would recommend using `map` for this instead. It's generally considered best practise for outputting a list of JSX elements. We've covered plenty of examples in class for you to reference, however, let me know if this isn't clear!

Reply:
Done. Thank you for your advice. You are right it was easy to do it and I don't know why I did not use map in the first place!